packages/server/src/fhir/repo.ts

```diff
@@ -216,12 +217,21 @@ export class Repository extends BaseRepository implements FhirRepository<PoolCli
     if (!this.context.author?.reference) {
       throw new Error('Invalid author reference');
     }
+
+    // Default to writer mode
+    // In the future, as we do more testing and validation, we will explore defaulting to reader mode
+    // However, for now, we default to writer and only use reader mode for requests guaranteed not to have consistency risks
+    this.mode = RepositoryMode.WRITER;
   }
 
   clone(): Repository {
     return new Repository(this.context);
   }
 
+  setMode(mode: RepositoryMode): void {
+    this.mode = mode;
```

Comment:
Given the comment below, should `setMode(RepositoryMode.READER)` throw if `getDatabaseClient` has already been called when `this.mode === RepositoryMode.WRITER`?

Reply:
Hm, interesting.

At present, the `mode` is initialized to `WRITER` (pessimistically).  We then set it to `READER` when we detect that it's a graphql request.

We could instead initialize to `undefined`?  Not sure that's much better? 🤷

Reply:
Yeah I think once the mode is set to `WRITER` it should probably be sticky. Initializing to `READER` but as soon as we see we are not in a "readonly gql query" we set it to `WRITER` where it is stuck for the rest of this Repository's lifetime might be want we want